src/Sulu/Bundle/ContentBundle/Repository/NodeRepository.php

```diff
@@ -108,7 +105,7 @@ public function __construct(
         $this->queryExecutor = $queryExecutor;
         $this->accessControlManager = $accessControlManager;
         $this->tokenStorage = $tokenStorage;
-        $this->logger = $logger;
```

Comment:
Why do we use now setter injection instead? I'd prefer constructor injection, and write this line as follows:

```php
$this->logger = $logger ?? new NullLogger();
```

With your current implementation we are always creating a `NullLogger` just to be replaced afterwards.

Reply:
@danrot I'm basically a friend of the `LoggerAwareInterface` and not having the logger in the constructor because I did see some strange constructor like this in the past:

```
__construct(
    SomeDependency $someDependency,
    LoggerInterface $logger = null,
    $parameter,
    $moreParameter
)
```

which basically will tell you in the IDE / code analyser optional parameters can not be in the middle of function, but if we say logger is always the last param of the constructor I'm fine with it.

Reply:
Then I would say we make it the last argument :slightly_smiling_face: